### PR TITLE
Qobuz provider tidy up and add tests

### PR DESCRIPTION
**Pull request by teancom**
A few "tidy up" stuff (typo of Qobux, an extra forward slash in a URL, some defensive programming for list access) and adding some ground-truth unit tests. I'm going through the Qobuz provider and wanted to make sure I wasn't going to break anything (as much as I can prevent that), so wanted to get tests in place to help with that.

The AI-generated summary:

Harden Qobuz API response parsing against missing/null fields:
- Guard against missing 'artist' key in get_artist_albums results
- Handle missing 'upc' field on albums without raising KeyError
- Fix performer field access (dict, not string) when filtering Various Artists
- Handle missing 'performers' key in track fallback artist resolution
- Skip malformed performer strings that lack role delimiter
- Fix indentation bug that appended artist outside conditional block
- Remove errant leading slash from reportStreamingEnd endpoint
- Fix "Qobux" typo in class docstring

Add comprehensive unit tests covering all parse methods (_parse_artist,
_parse_album, _parse_track, _parse_playlist) with edge cases for missing
fields, Various Artists filtering, and playlist editability.

### CODE REVIEW

File: music_assistant/providers/qobuz/__init__.py [hunk 1]
```diff
@@ -683,22 +689,24 @@ async def _parse_track(self, track_obj: dict[str, Any]) -> Track:
         )
         if isrc := track_obj.get("isrc"):
             track.external_ids.add((ExternalID.ISRC, isrc))
-        if track_obj.get("performer") and "Various " not in track_obj["performer"]:
+        if track_obj.get("performer") and "Various" not in track_obj["performer"].get("name", ""):
```
Comment (MarvinSchenkel) [inline music_assistant/providers/qobuz/__init__.py]: Not sure if removing the space is the right approach here? Then we would also match "Variouslywhatever" ?

File: tests/providers/qobuz/test_parsers.py [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+        assert result.metadata.images is not None
+        assert len(result.metadata.images) == 1
+
+    async def test_parse_track_performer_various_artists(
+        self, mock_provider: QobuzProvider
+    ) -> None:
+        """Test parsing track with Various Artists performer is filtered."""
+        track_with_various = deepcopy(TRACK_OBJ)
+        track_with_various["performer"] = {
+            "id": 145383,
```
Comment (MarvinSchenkel) [inline tests/providers/qobuz/test_parsers.py]: Please use the constant from the init for all these various artist IDs

File: music_assistant/providers/qobuz/__init__.py (hunk 1 above)
Comment (teancom) [reply music_assistant/providers/qobuz/__init__.py]: Great catch, I had noted that as probably not being right and then got distracted and never came back. Updated this and one other place where we could be using the constant instead and stop parsing strings. 🙇

**Comment by marcelveldt** [comment]
I'm assuming you tested the Qobuz provider with your own account after these changes ?

**Comment by teancom** [comment]
Yep, for the last few days.

**Review by MarvinSchenkel — APPROVED** [review]
Thanks @teancom 👍
